.github/workflows/clippy-async.yml

```diff
@@ -0,0 +1,19 @@
+on:
+  push:
+    branches: [ staging, trying, master ]
+  pull_request:
+
+name: Clippy check
+jobs:
+  clippy_check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          profile: minimal
+          toolchain: nightly
+          override: true
+          components: clippy
+      - run: cargo clippy
+        working-directory: embedded-nal-async
```

Comment:
Similar to above - please merge these checks as a separate job (or matrix job) in `clippy.yml`